cni/network/network_windows.go

```diff
@@ -177,6 +177,15 @@ func (plugin *NetPlugin) getNetworkName(podName, podNs, ifName string, nwCfg *cn
 			return networkName, err
 		}
 
+		// First try to get the network name from the state file
+		if networkName, err := plugin.nm.FindNetworkNameFromNetNs(netNs); err != nil {
+			log.Printf("Error getting network name from state: %v. Fall back to CNS.", err)
+		} else {
+			return networkName, nil
+		}
+
+		// If it is not found, then fallback to CNS, in case it wasn't in the statefile for some reason
+		// TODO: investigate if we can remove this fallback
 		_, cnsNetworkConfig, _, err = plugin.multitenancyClient.GetContainerNetworkConfiguration(context.TODO(), nwCfg, podName, podNs, ifName)
```

Comment:
i would say remove it. even if it fails, we know whether to fix this or not

Reply:
actually I think we need this, during CNI ADD it will call this function too and network won't be in state file yet since it hasn't been created, so in that case we need to query CNS

Reply:
hmm right.. we are calling CNS twice in ADD case then.. we should remove this code and rely on GetMultiTenancyCNIResult which returns cnsNetworkConfig too and we have to construct from that

Reply:
done